VideoLocker/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -239,60 +239,62 @@ public ProgressBar getProgressBar() {
-                                LoginErrorMessage errorMsg =
-                                        new LoginErrorMessage(
-                                                getString(R.string.login_error),
-                                                getString(R.string.login_failed));
-                                throw new LoginException(errorMsg);
-                            }
-                        } catch (LoginException ex) {
-                            super.onException(ex);
-                            onUserLoginFailure(ex, null, null);
+        if (!NetworkUtil.isConnected(this)) {
+            showErrorMessage(getString(R.string.no_connectivity),
+                    getString(R.string.network_not_connected));
+            return;
+        }
+
+        emailStr = email_et.getText().toString().trim();
+        String passwordStr = password_et.getText().toString().trim();
+
+        if (email_et != null && emailStr.length() == 0) {
+            showErrorMessage(getString(R.string.login_error),
+                    getString(R.string.error_enter_email));
+            email_et.requestFocus();
+        } else if (password_et != null && passwordStr.length() == 0) {
+            showErrorMessage(getString(R.string.login_error),
+                    getString(R.string.error_enter_password));
+            password_et.requestFocus();
+        } else {
+            email_et.setEnabled(false);
+            password_et.setEnabled(false);
+            forgotPassword_tv.setEnabled(false);
+            eulaTv.setEnabled(false);
+
+            clearDialogs();
+
+            LoginTask logintask = new LoginTask(this, email_et.getText().toString().trim(),
+                    password_et.getText().toString()) {
+                @Override
+                public void onSuccess(AuthResponse result) {
+                    try {
+                        if (result != null && result.hasValidProfile()) {
+                            onUserLoginSuccess(result.profile);
+                        } else {
+                            LoginErrorMessage errorMsg =
+                                    new LoginErrorMessage(
+                                            getString(R.string.login_error),
+                                            getString(R.string.login_failed));
+                            throw new LoginException(errorMsg);
                         }
-                    }
-                    @Override
-                    public void onException(Exception ex) {
+                    } catch (LoginException ex) {
                         super.onException(ex);
-                        tryToSetUIInteraction(true);
+                        onUserLoginFailure(ex, null, null);
                     }
+                }
 
-                };
+                @Override
+                public void onException(Exception ex) {
+                    super.onException(ex);
+                    tryToSetUIInteraction(true);
```

Comment:
`onUserLoginFailure` would have to be refactored to take any exception which I'm not sure is what we want? If we set `onUserLoginFailure(ex, null, null)` it does the same thing as `tryToSetUIInteraction(true);` 

More discussion here https://github.com/edx/edx-app-android/pull/637